### PR TITLE
suse: Fix path to dracut dir

### PR DESCRIPTION
**Pull request by gmbr3**
It should be /usr/lib/dracut not /usr/libexec/dracut

### CODE REVIEW

**Comment by rleon** [comment]
Please add your "Signed-of-by: ..." tag and the commit message which describes the change.

@nmorey it is worth to fix before we will release rdma-core (next week. on -rc1).

**Comment by nmorey** [comment]
Pretty sure the fix is incomplete.
This fixes the change of %{_libexec} from /usr/lib to /usr/libexec
It does fix the dracut file install path but not its content.
The path to mlx4-setup.sh is hardcoded in it
`        inst /usr/lib/mlx4-setup.sh`
But the specfile installs it in libexec:
`install -D -m0755 redhat/rdma.mlx4-setup.sh %{buildroot}%{_libexecdir}/mlx4-setup.sh`
@gmbr3: There's also this to fix (which you have fixed in your OBS project but not in this PR):
`sed 's%/usr/libexec%/usr/lib%g' redhat/rdma.mlx4.sys.modprobe > %{buildroot}%{sysmodprobedir}/50-libmlx4.conf`

@rleon With the couple things I mentioned fix, it'd probably be good to add to the next release. Anyway it's just packaging issue for Tumbleweed so worst case, I can just fix it in OBS if it's too late for you to add.
It'll also have to be backported to pretty much all stable branches

**Comment by gmbr3** [comment]
changes done

File: suse/rdma-core.spec [hunk 1]
```diff
@@ -452,14 +452,19 @@ mkdir -p %{buildroot}%{sysmodprobedir}
 mkdir -p %{buildroot}%{_unitdir}
 
 # Port type setup for mlx4 dual port cards
+install -Dm0644 redhat/rdma.mlx4.sys.modprobe %{buildroot}%{sysmodprobedir}/50-libmlx4.conf
```
Comment (rleon) [inline suse/rdma-core.spec:455]: "-Dm0644" -> "-D -m0644"

**Review by rleon — COMMENTED** [review]
Please add this line above your SOB tag, thanks

Fixes: 8854eefa60b5 ("suse: Add spec file for SUSE")

**Comment by rleon** [comment]
@nmorey, I'm confident that we will finish the review before release date.

Generally, I'm releasing on -rc1, but for major fixes (API, packaging, build e.t.c) can delay upto -rc2.

Thanks

File: suse/rdma-core.spec [hunk 2]
```diff
@@ -452,14 +452,19 @@ mkdir -p %{buildroot}%{sysmodprobedir}
...
 install -D -m0644 redhat/rdma.mlx4.conf %{buildroot}/%{_sysconfdir}/rdma/mlx4.conf
-sed 's%/usr/libexec%/usr/lib%g' redhat/rdma.mlx4.sys.modprobe > %{buildroot}%{sysmodprobedir}/50-libmlx4.conf
```
Comment (nmorey) [inline suse/rdma-core.spec:456]: Please keep the sed but use %_libexec
We need these to work on existing SLES which are still using libexec as /usr/lib

File: suse/rdma-core.spec [hunk 3]
```diff
@@ -452,14 +452,19 @@ mkdir -p %{buildroot}%{sysmodprobedir}
...
 install -D -m0755 redhat/rdma.mlx4-setup.sh %{buildroot}%{_libexecdir}/mlx4-setup.sh
 
 # Dracut file for IB support during boot
 install -D -m0644 suse/module-setup.sh %{buildroot}%{dracutlibdir}/modules.d/05rdma/module-setup.sh
 
+%if "%{_libexecdir}" != "/usr/libexec"
+sed 's-/usr/libexec-%{_libexecdir}-g' -i %{buildroot}%{sysmodprobedir}/50-libmlx4.conf
+sed 's-/usr/libexec-%{_libexecdir}-g' -i %{buildroot}%{dracutlibdir}/modules.d/05rdma/module-setup.sh
+%endif
+
```
Comment (nmorey) [inline suse/rdma-core.spec:467]: Ok this solves the previous remark I made.

**Comment by gmbr3** [comment]
changes done

**Comment by rleon** [comment]
Please remove blank line between Fixes and SOB, thanks.

""
Fixes: 8854eef ("suse: Add spec file for SUSE")

Signed-off-by: Callum Farmer <gmbr3@opensuse.org>
""
-->
""
Fixes: 8854eef ("suse: Add spec file for SUSE")
Signed-off-by: Callum Farmer <gmbr3@opensuse.org>
""

**Comment by gmbr3** [comment]
done

**Comment by nmorey** [comment]
Thanks

@rleon on There's an issue with the tumbleweed docker/cbuild at the moment so I can't fully test this out but it looks good to me

**Pull request merged**